### PR TITLE
`TextCatParametricAttention.v1`: set key transform dimensions

### PR DESCRIPTION
**Pull request by danieldk**
This is necessary for tok2vec implementations that initialize lazily (e.g. curated transformers).

## Description

Set key transform dimensions in `TextCatParametricAttention.v1`, otherwise the key transformation dimensionality cannot be inferred for lazily-initialized tok2vecs listeners like transformers.

Not sure how we can test this without making a new tok2vec model in spaCy that is lazily initialized (does know its width until initialization). Maybe we should add a separate model to the tests?

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Review by svlandeg — COMMENTED** [review]
Looks good.

Would it be much work to make a sort of mockup `tok2vec` in the test suite for this purpose? I do think it's the sort of thing where we should have better tests because these things may come to bite us again in the future...

**Comment by danieldk** [comment]
> Would it be much work to make a sort of mockup `tok2vec` in the test suite for this purpose? I do think it's the sort of thing where we should have better tests because these things may come to bite us again in the future...

Agreed, I'll give it a try. Might reveal issues in the other implementations as well.

**Comment by danieldk** [comment]
Added, this uncovered two more issues.

**Comment by danieldk** [comment]
Tests break currently because of the issue that #13284 fixes.

**Comment by danieldk** [comment]
Ready for another review round.

File: spacy/tests/tok2vec.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from typing import List
+
+from thinc.api import Model
+from thinc.types import Floats2d
+
+from spacy.tokens import Doc
+from spacy.util import registry
+
+
+@registry.architectures("LazyInitTok2Vec.v1")
```
Comment (svlandeg) [inline spacy/tests/tok2vec.py:10]: Maybe we should add a "test." prefix just to be fully clear this isn't meant to be used in any way?

**Pull request merged**